operator-common/src/main/java/io/strimzi/operator/common/metrics/AbstractMetricsHolder.java

```diff
@@ -0,0 +1,230 @@
+/*
+ * Copyright Strimzi authors.
+ * License: Apache License 2.0 (see the file LICENSE or http://apache.org/licenses/LICENSE-2.0.html).
+ */
+package io.strimzi.operator.common.metrics;
+
+import io.micrometer.core.instrument.Counter;
+import io.micrometer.core.instrument.Tag;
+import io.micrometer.core.instrument.Tags;
+import io.micrometer.core.instrument.Timer;
+import io.strimzi.operator.common.MetricsProvider;
+import io.strimzi.operator.common.model.Labels;
+
+import java.util.Map;
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.function.Function;
+
+/**
+ * Util class which holds the different metrics used by controllers
+ */
+public abstract class AbstractMetricsHolder {
```

Comment:
IDK whether we really need to put `Abstract` in the class name. I.e. why not just `MetricsHolder`?